services/src/api/handlers/layers.rs

```diff
@@ -314,6 +324,116 @@ async fn list_collection_handler<C: ApplicationContext>(
     Ok(web::Json(collection))
 }
 
+async fn search_capabilities_handler<C: ApplicationContext>(
+    app_ctx: web::Data<C>,
+    path: web::Path<DataProviderId>,
+    session: C::Session,
+) -> Result<impl Responder> {
+    let provider = path.into_inner();
+
+    if provider == ROOT_PROVIDER_ID {
+        return Err(NotImplemented {
+            message: "Global search is not implemented".to_string(),
+        });
+    }
+
+    let db = app_ctx.session_context(session).db();
+
+    if provider == crate::datasets::storage::DATASET_DB_LAYER_PROVIDER_ID.into() {
+        return Err(NotImplemented {
+            message: "Dataset search is not implemented".to_string(),
+        });
+    }
```

Comment:
Yeah, I can change that. I copied those if statements from the collection listing handler, no other reason for it being that way

Reply:
https://github.com/geo-engine/geoengine/pull/900/commits/6b48194bfcd5e7302013d2cd8d8f984eab28097c

Changed it for the dataset providers. For the "root provider" though I now see why I did it like this. There is no actual "root provider" trait where I could add a search/autocomplete method. I could do it here in the handler like the `getLayerProviders` method, but it would not be better than it is right now I think.

Reply:
Looks good.
Maybe you can also use the term "supported" here instead of implemented.

Reply:
https://github.com/geo-engine/geoengine/pull/900/commits/d5554082076c5756275c9258a0b6d7922147b4fe

Also forgot to add the capabilites to the dataset provider in the previous commit https://github.com/geo-engine/geoengine/pull/900/commits/3f63b77ca57204c0410c9fa6ecaef79ff0fc6f54